### PR TITLE
moved getPopulate() logic to load()

### PR DESCRIPTION
**Pull request by eahefnawy**
we no longer write/save ... so we don't need to keep the unpopulated data as suggested by @ac360

### CODE REVIEW

File: lib/classes/Service.js [hunk 1]
```diff
@@ -44,82 +44,81 @@ class Service {
...
+          varTemplateSyntax = RegExp(this.variableSyntax, 'g');
+
+          // temporally remove variable syntax from service otherwise it'll match
+          this.variableSyntax = true;
+        }
+
+        const variablesObject = this.getVariables(options.stage, options.region);
+
+
+        // temporally remove environment obj. Doesn't make sense to
```
Comment (flomotlik) [inline lib/classes/Service.js]: temporally -> temporarily

**Comment by flomotlik** [comment]
@eahefnawy looks good to me, though I would like to have @ac360 to have a quick look as he should have more insight into this codebase as well. @ac360 let us know if this is fine for you too then we'll merge.

**Comment by ac360** [comment]
Thanks, @eahefnawy.  A few questions:

* This only parses YAML and not JSON files.  Did we make a decision not to deal w/ `serverless.json` files?
* If we are allowing YAML and JSON files, `YamlParser` should be renamed to `Parser`.  Or you could get rid of this class altogether, and make it a simple utility method.
* Why are we still using `serverless.instances`?

**Comment by eahefnawy** [comment]
@ac360 my understanding is that the main config file will only be a yaml file (serverless.yaml) ... and from this yaml file can parse references to other json/yaml files using json-ref.

It used to be a utility method but @flomotlik suggested to wrap it into a class for separation and keeping the utility file small. I personally prefer a simple utility method IMHO.

hmmm I don't think we had an explicit decision to drop `instances` ... I personally don't like it either, but I'm keeping it for the sake of being explicit like the rest of our codebase

**Comment by flomotlik** [comment]
> simple utility method

I think architecturally its much cleaner to keep it in its own class. Utilities really should only have small things that don't fit somewhere else as you otherwise have larger Utility Files and especially test files around that are hard to understand.

>hmmm I don't think we had an explicit decision to drop instances ... I personally don't like it either, but I'm keeping it for the sake of being explicit like the rest of our codebase

Imho we can remove it.

**Pull request merged**